### ISSUE TITLE
Feature request: Export and Import containers & settings

### ISSUE THREAD

**Issue by rm-root** [mozilla/multi-account-containers#1282]
Hi, will you develop a way to export and import containers and settings? At the moment I must setup a lot of containers every time I use conteiners on a new device. Thank you.

**Comment by IOAyman** [comment]
That would be great!

**Comment by alex3kov** [comment]
Please do this. Preferably export everything (containers and URLs assigned to them) to one file - json or whatever.

**Comment by pierlauro** [comment]
Waiting for an eventual official support, I created a script to export and import containers among different machines.

https://github.com/pierlauro/ffcontainers

**Comment by MathieuDR** [comment]
> 
> 
> Waiting for an eventual official support, I created a script to export and import containers among different machines.
> 
> https://github.com/pierlauro/ffcontainers

Sadly doesn't work for me. But I used it to manually export it, Lets see if I can import it as well.
Thanks!

**Comment by pierlauro** [comment]
> 
> 
> > Waiting for an eventual official support, I created a script to export and import containers among different machines.
> > https://github.com/pierlauro/ffcontainers
> 
> Sadly doesn't work for me. But I used it to manually export it, Lets see if I can import it as well.
> Thanks!

Please, open an issue on [ffcontainers' issue tracker](https://github.com/pierlauro/ffcontainers/issues) detailing the problem so I will be able to fix it.

**Comment by throwaway167** [comment]
Adding an import & export capability for Containers, Settings, & Container's URL lists would be a TREMENDOUS asset. — As someone who uses profiles across multiple PCs, and occasionally has to delete & recreate FF profiles. . . needing to **manually recreate all your containers and their associated URLs** can be _extremely time-consuming_ and a complete pain. 

Would be MUCH appreciated, thank you!

**Comment by avliet** [comment]
Import/Export would make life a LOT easier when upgrading Firefox on my Debian hosts.  Right now every time I upgrade Firefox, I lose all my containers, sessions, everything. Not this this project's fault, but REALLY annoying.

**Comment by drod3763** [comment]
Same right here especially now that sync isn't working for some people. - myself included. This would be a good workaround.

**Comment by moocan** [comment]
Hello,

I upvote for this too.
It would be very useful to have an `export/import settings` feature using a file in `Multi-Account Containers` like some add-ons allow you to do it already.

I use a lot of containers because I have a dedicated container for each site that requires authentication and temporary containers for all other sites, but I have to manually recreate them every time I reset my profile or my operating system. I can export and import many addon settings quickly, but not from this add-on.

I'm sorry, but not everyone (or maybe just me :( ) synchronizes their life over Internet too.

Kind Regard

**Comment by Jieiku** [comment]
I just temporarily turned off this addon to see if it was the cause of a problem I was having, now all of my custom containers are gone, really need a feature like this.

**Comment by gelatinbomb** [comment]
I completely agree with you. Especially if you have a lot of containers and you want to factory reset your pc.

**Comment by Dormann** [comment]
Just like @moocan : I have to reinstall my new Debian and do not wish to sync over internet.
I do everything on local and so would I like doing for this (super useful) add-on.
Please add local export / import.

**Comment by kocjs** [comment]
All other add-ons allow to export their settings to .json, but Mozilla decided lock us into Firefox sync with their only add-on. This tells the story, make it easy for everyone, but hard for the view.

**Comment by Gitoffthelawn** [comment]
This is likely the most requested feature request for this extension.  Is anyone interested in implementing this feature?  Perhaps we can take up a collection to get it done.

**Comment by ralpha** [comment]
I was looking for a solution for exporting this data too.
I did not write a script to import the data. But this should not be to difficult. After using the info below.

Go to: `about:debugging#/runtime/this-firefox` (put this in URL bar in Firefox)

Go to `Firefox Multi-Account Containers` and click `inspect`
<img width="676" height="110" alt="Image" src="https://github.com/user-attachments/assets/98b122f2-d2cc-4684-8aa7-db5ed49a94de" />

Go to Storage -> Extension Storage -> `moz-extension://...`
There you see all the data.
<img width="1222" height="444" alt="Image" src="https://github.com/user-attachments/assets/8075d6bc-8b47-4195-adc2-15728b341707" />


To export use the `Console`: (needs to be in same windows as above! Otherwise the data is not accessible)
```js
console.log(JSON.stringify(await browser.storage.local.get(), null, 2));
```

Copy output and store it in a file somewhere.

Container numbers correspond to `~/.mozilla/firefox/<profile_name>/containers.json`

To see where it loops over the objects in the Git repo: https://github.com/mozilla/multi-account-containers/blob/aca51cc11c85973e75d89002171c5bc37393ea22/src/js/background/assignManager.js#L109-L129
This can be used to `.set()` all the objects back in place. (and thus import)

I hope this gets someone motivated to implement this in the UI.

**Comment by Gitoffthelawn** [comment]
> I was looking for a solution for exporting this data too. ...

Thanks!  Of course once someone has the data, import functionality is needed too!

**Comment by shadow-absorber** [comment]
sad that this is not implemented yet... really looks like a no brainer that this is a feature people want and is very helpful and useful
